cmake/checks/check_02_compiler_features.cmake

```diff
@@ -407,3 +407,25 @@ if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 
   reset_cmake_required()
 endif()
+
+
+#
+# Check whether the compiler supports interprocedural and link-time
+# optimization. If so, we can set the corresponding property on the
+# compile and link targets later on.
+#
+if(DEAL_II_USE_LTO)
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT _res OUTPUT _out LANGUAGES CXX)
+  if (_res)
+    message(STATUS "Compiler supports interprocedural/link-time optimizations")
+    set(DEAL_II_USE_LTO "YES")
+  else()
+    message(FATAL_ERROR "You asked for interprocedural/link-time optimizations, but the compiler does not support these optimizations: ${_out}")
+    set(DEAL_II_USE_LTO "NO")
+  endif()
+else()
+  # User did not set DEAL_II_USE_LTO on the command line. Set the
+  # variable to a specific (negative) value.
+  set(DEAL_II_USE_LTO "NO")
+endif()
```

Comment:
```suggestion
if(DEAL_II_USE_LTO)
  include(CheckIPOSupported)
  check_ipo_supported(RESULT _res OUTPUT _out LANGUAGES CXX)
  if (_res)
    message(STATUS "Compiler supports interprocedural/link-time optimizations")
  else()
    message(FATAL_ERROR "You asked for interprocedural/link-time optimizations, but the compiler does not support these optimizations: ${_out}")
  endif()
endif()
```

Reply:
See suggestion: feel free to clean up the logic a bit more. Or we can merge as is and clean up later.

Reply:
I think what you're saying is that because the variable is a "truthy" one, we don't actually specifically have to set it to YES and NO, or any other value. Yes. I'll fix that in a follow-up.